### DISCUSSION TITLE
Does Matter integration allow less than 10 devices per ESP32 board

### DISCUSSION THREAD

**Discussion by ivanvachovsky** [arendst/Tasmota#22286] [Q&A]
Running Tasmota [Tasmota 14.2.0 (release-tasmota32) by Theo Arends](https://bit.ly/tasmota) on ESP32 board. I do have 12 relays set up. I noticed that Matter will make only 9 (the ones with single digit) available in Goggle Home. Is this normal ? If yes than does the restriction apply to same type devices. Say if I have 9 relays than can I add 9 PWM ?

**Comment by s-hadinger** [comment]
I have never tested up to 12 devices which may become stretch with RAM. Is your device serial connected? If so set log level to `seriallog 3` and add `"debug":true,` in the Matter configuration json (make sure to write a valid json file). Then pair again and paste the full logs. It will be quite long
